UPGRADING.md

```diff
@@ -218,111 +88,64 @@ domain.Create(context.Background(), &domain.CreateParams{
     IsSatellite: clerk.Bool(true),
 })
 ```
-
 You can explicitly pass zero values with `clerk.String("")` or `clerk.Int64(0)`.
 
-All API operations receive a `context.Context` as the first argument.
-
-## HTTP middleware
-
-The `v1` version of the Clerk Go SDK supports two HTTP middleware functions that can handle authentication with Clerk.
-These are `WithSessionV2` and `RequireSessionV2`.
-
-The middleware in `v1` supports authentication with a bearer token in the HTTP request headers, falling back to cookie
-based authentication if the "Authorization" header is missing.
-
-In a similar way, the `v2` version of the library also provides two middleware functions that can handle authentication
-with Clerk; `WithHeaderAuthorization` and `RequireHeaderAuthorization`.
-
-As the name implies, the new middleware support only authentication with a bearer token, that needs to be present in the
-"Authorization" header of the HTTP request.
-
-Cookie based authentication is not supported at all by the `v2` version of the library.
-
-Usage has also changed between `v1` and `v2`, as have the different options that the middleware support.
-
-In `v1`, here's how you would use the `RequireSessionV2` middleware to get access to the session token claims in HTTP handlers.
-
-```go
-import (
-    "net/http"
-
-    "github.com/clerkinc/clerk-sdk-go"
-)
-
-func main() {
-    client, err := clerk.NewClient("sk_live_XXX")
-    if err != nil {
-        panic(err)
-    }
-    mux := http.NewServeMux()
-    mux.Handle("/session", clerk.RequireSessionV2(client)(http.HandlerFunc(handleSession)))
-    http.ListenAndServe(":3000", mux)
-}
-
-func handleSession(w http.ResponseWriter, r *http.Request) {
-    sessionClaims, ok := clerk.SessionFromContext(r.Context())
-    if ok {
-        // claims contain session information
-    } else {
-        // there is no active session (non-authenticated user)
-    }
-}
+### The type for failed API responses is now `*clerk.APIErrorResponse`
+
+The `v2` version of the library introduces a new type for API responses that contain errors.
+The new type is [clerk.APIErrorResponse](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2#APIErrorResponse) and it replaces `clerk.ErrorResponse`.
+
+The `clerk.APIErrorResponse` type contains additional fields and provides access to more debugging information.
+
+```diff
+- org, err := client.Organizations().Create(clerk.CreateOrganizationParams{
+-     Name: "Acme Inc",
+- })
+- if err != nil {
+-     if errResp, ok := err.(*clerk.ErrorResponse); ok {
+-         // Access the API errors
+-         errResp.Errors
+-     }
+- }
++ ctx := context.Background()
++ clerk.SetKey("sk_live_XXX")
++ org, err := organization.Create(ctx, &organization.CreateParams{
++     Name: clerk.String("Acme Inc"),
++ })
++ if err != nil {
++     if apiErr, ok := err.(*clerk.APIErrorResponse); ok {
++         // Access the API errors and additional information
++         apiErr.TraceID
```

Comment:
💡 You can maybe emulate logging here, so that this section becomes valid Go as well:

```go
fmt.Printf("[%s] request failed: %s (%s)\n", 
  apiErr.TraceID,   
  apiErr.Error(), 
  string(apiErr.Response.RawJSON))
```

Reply:
Nice! I think I'll change it altogether to remove code and focus on just the type change.

The problem with Go is that it's too verbose and it's hard to draw the line of how many details to add while preserving readability on the change in focus.